### PR TITLE
ci(netlify): add netlify deploy workflows

### PR DESCRIPTION
**Pull request by ismay**
This migrates the netlify deploy workflows that we've been testing in the data entry app (https://github.com/dhis2/aggregate-data-entry-app) to our generic workflows.

I've renamed the dhis2-preview-pr workflow to match the naming scheme of the other workflows, for clarity. This means that anyone migrating to these new workflows will have to remove the old one manually, if it exists. That should not be too big of a deal though, it's just deleting a single file.

- https://dhis2.atlassian.net/browse/LIBS-389
- https://github.com/dhis2/aggregate-data-entry-app/pull/278

### CODE REVIEW

**Review by Birkbjo — APPROVED** [review]

File: ci/dhis2-netlify-deploy-pr.yml [hunk 1]
```diff
@@ -17,14 +17,14 @@ concurrency:
...
-    preview:
+    deploy:
         runs-on: ubuntu-latest
         if: "!github.event.push.repository.fork && github.actor != 'dependabot[bot]'"
         steps:
             - uses: actions/checkout@v2
             - uses: actions/setup-node@v1
               with:
-                  node-version: 12.x
+                  node-version: 14.x
```
Comment (Birkbjo) [inline ci/dhis2-netlify-deploy-pr.yml:27]: We're using node 16 above, should we use the same version?

Comment (ismay) [reply ci/dhis2-netlify-deploy-pr.yml:27]: Yeah you're right. 16 or even 18 would be good. I'll update that.

File: ci/dhis2-netlify-deploy-branch.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+concurrency:
+    group: ${{ github.workflow}}-${{ github.ref }}
+    cancel-in-progress: true
+
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (ismay) [reply ci/dhis2-netlify-deploy-branch.yml:26]: No reason, you're right, would be good to upgrade. I'll take a look at that.

File: ci/dhis2-netlify-deploy-branch.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            - name: Deploy
+              id: netlify-deploy
+              uses: nwtgck/actions-netlify@v1.2.4
+              timeout-minutes: 1
+              with:
+                  github-token: ${{ secrets.DHIS2_BOT_GITHUB_TOKEN }}
+                  github-deployment-environment: 'netlify'
+                  github-deployment-description: 'Branch Deploy on Netlify'
+                  deploy-message: Github actions branch deploy
+                  enable-pull-request-comment: false
```
[the remark this answers is not in the document]
Comment (ismay) [reply ci/dhis2-netlify-deploy-branch.yml:46]: We already do this, but only for the PR preview, see here: https://github.com/dhis2/workflows/blob/b71188de020dfa47febe63c828a1a0e97e90516a/ci/dhis2-netlify-deploy-pr.yml#L44

The branch deploy isn't necessarily associated with a PR, so I'm assuming that this setting has no effect for a branch deploy.

[the remark this answers is not in the document]
Comment (ismay) [reply ci/dhis2-netlify-deploy-branch.yml:46]: Took a look, from the source code it looks like it only has an effect if it's a PR: https://github.com/nwtgck/actions-netlify/blob/af8e90039cc437e20c585f6d0a1c3cc93d04ef12/src/main.ts#L155

File: ci/dhis2-netlify-deploy-production.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            - name: Deploy
+              id: netlify-deploy
+              uses: nwtgck/actions-netlify@v1.2.4
+              timeout-minutes: 1
+              with:
+                  github-token: ${{ secrets.DHIS2_BOT_GITHUB_TOKEN }}
+                  github-deployment-environment: 'netlify'
+                  github-deployment-description: 'Production Deploy on Netlify'
+                  deploy-message: Github actions production deploy
+                  enable-pull-request-comment: false
```
[the remark this answers is not in the document]
Comment (ismay) [reply ci/dhis2-netlify-deploy-production.yml:47]: See my comment on the other wf.

**Pull request merged**